proposals/debug-id.md

```diff
@@ -0,0 +1,237 @@
+   will make it easier to identify and associate minified JavaScript files with
+   their corresponding source maps.
+
+2. Self-Identifying Files: This specification changes source maps and minified
+   JavaScript files so that they become self-identifying, eliminating the need
+   for external information to work with the files.
+
+3. Streamlined Debugging Process: The implementation of Debug IDs will simplify
+   and streamline the debugging process by reducing the likelihood of errors
+   resulting from misidentification or misassociation of files.
+
+4. Standardization: The adoption of this proposal as a web standard will
+   encourage a consistent and unified approach to handling source maps and
+   minified JavaScript files across the industry.
+
+5. Guaranteed bidirectionality: Today source maps do not provide the ability to
+   reliably resolve back to the transpiled file they are from.  However in
+   practice tools often require this as they are often leveraging the
+   transpiled artifact to resolve scope information by parsing the source.
+
+6. Symbol server support: with Debug IDs and source maps with embedded sources
+   it becomes possible to support symbol server lookup from symbol servers.
+
+## Scope
+
+This proposal sets some specific limitations on source maps to simplify the
+processing in the wider ecosystem.  Debug IDs are at present only specified to
+source maps with embedded sources or where sources are categorically not
+available.  The lookup for original sources from a source map identified by a
+debug ID is not defined.
+
+Additionally, this specification applies only to non-indexed source maps and
+currently specifies references only for JavaScript.
+
+## Terms
+
+In the context of this document:
+
+- **Source Map:** Refers to a non-indexed, standard source map.
+- **Transpiled File:** Refers to a transpiled (potentially minified) JavaScript file.
+- **Debug ID:** Refers to a UUID as described in this document.
+
+## Debug IDs
+
+Debug IDs are globally unique identifiers for build artifacts.  They are
+specified to be UUIDs.  In the context of this proposal, they are represented in
+hexadecimal characters.  When comparing debug IDs they must be normalized.  This
+means that `85314830-023f-4cf1-a267-535f4e37bb17` and
+`85314830023F4CF1A267535F4E37BB17` are equivalent but the former representation
+is the canonical format.
+
+The way a debug ID is generated is specific to the toolchain and no requirements
+are placed on it.  It is however recommended to generate deterministic debug IDs
+(UUID v3 or v5) so that rebuilding the same artifacts yields stable IDs.
+
+Debug IDs are embedded in both source maps and transpiled files, allowing a
+bidirectional mapping between them. The linking of source maps and transpiled
+files via HTTP headers is explicitly not desired.  A file identified by a Debug
+ID must have that Debug ID embedded to ensure the file is self-identifying.
```

Comment:
Not sure if this needs to be mentioned here or anywhere. Tools like webpack have an option to have `"hidden"` (See https://webpack.js.org/configuration/devtool/#devtool) sourcemaps, which are created but not referenced, to create some sense of "obfuscation". As the Debug ID would just be some random bytes, they do not pose any "obfuscation" risk so to say, and tools should always put them in.